puppetwf/activity_test.go

```diff
@@ -0,0 +1,165 @@
+package puppetwf_test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"github.com/lyraproj/pcore/px"
+	"github.com/lyraproj/pcore/types"
+	"github.com/lyraproj/puppet-evaluator/pdsl"
+	"github.com/lyraproj/puppet-evaluator/puppet"
+	"github.com/lyraproj/puppet-workflow/puppetwf"
+	"github.com/lyraproj/servicesdk/grpc"
+	"github.com/lyraproj/servicesdk/serviceapi"
+	"github.com/stretchr/testify/assert"
+)
+
+func withSampleService(sf func(pdsl.EvaluationContext, serviceapi.Service)) {
+	puppet.Do(func(ctx pdsl.EvaluationContext) {
+		// Command to start plug-in and read a given manifest
+		cmd := exec.Command("go", "run", "../main.go", "--debug")
+
+		// Logger that prints JSON on Stderr
+		logger := hclog.New(&hclog.LoggerOptions{
+			Level:           hclog.Debug,
+			Output:          os.Stderr,
+			JSONFormat:      false,
+			IncludeLocation: false,
+		})
+
+		server, err := grpc.Load(cmd, logger)
+
+		// Ensure that plug-ins die when we're done.
+		defer func() {
+			wait := make(chan bool)
+			go func() {
+				plugin.CleanupClients()
+				wait <- true
+			}()
+
+			select {
+			case <-wait:
+			case <-time.After(2 * time.Second):
+			}
+		}()
+
+		if err == nil {
+			sf(ctx, server)
+		} else {
+			fmt.Println(err)
+		}
+	})
+}
+
+/*
+func withSampleLocalService(sf func(pdsl.EvaluationContext, serviceapi.Service)) {
+	pwf.WithService(`Puppet`, sf)
+}
+*/
+
+func TestActivity(t *testing.T) {
+	withSampleService(func(ctx pdsl.EvaluationContext, s serviceapi.Service) {
+		s.Metadata(ctx)
+		rs := s.Invoke(ctx, puppetwf.ManifestLoaderID, "loadManifest", types.WrapString("testdata"), types.WrapString("testdata/aws_example.pp")).(serviceapi.Definition)
+		v := s.Invoke(ctx, rs.Identifier().Name(), "metadata")
+		assert.Implements(t, (*px.List)(nil), v, "metadata type")
```

Comment:
Could be worth using [require](https://godoc.org/github.com/stretchr/testify/require) instead of assert here?